Streamline documentation to reflect project "piper" (#875)
SAP/jenkins-library: documentation/docs/extensibility.md

>   You have to create a file like `<StageName>.groovy` for example `Acceptance.groovy` and store it in folder `.pipeline/extensions/` in your source code repository.

Reviewer: ```suggestion
  You have to create a file like `<StageName>.groovy` (for example, `Acceptance.groovy`) and store it in folder `.pipeline/extensions/` in your source code repository.
```

!!! note "Cloud SDK Pipeline"

>     If you're using Cloud SDK Pipeline, the folder is named `pipeline/extensions/` (without the dot). Refer to [the Cloud SDK Pipeline documentation](https://github.com/SAP/cloud-s4-sdk-pipeline/blob/master/doc/pipeline/extensibility.md) for details.

Reviewer: ```suggestion
    If you use the Cloud SDK Pipeline, the folder is named `pipeline/extensions/` (without the dot). For more information, please refer to [the Cloud SDK Pipeline documentation](https://github.com/SAP/cloud-s4-sdk-pipeline/blob/master/doc/pipeline/extensibility.md).
```

>   The pipeline template will check if such a file exists and executes it if present.

Reviewer: ```suggestion
  The pipeline template checks if such a file exists and executes it, if present.
```

>   A parameter is passed to the extension containing following keys:

Reviewer: ```suggestion
  A parameter that contains the following keys is passed to the extension:
```
